### ISSUE TITLE
Timestamp: Unable to read timestamp written by gocql

### ISSUE THREAD

**Issue by QuentinPerez** [scylladb/scylla-rust-driver#316]
👋 everyone, 

The empty timestamps written by the gocql driver can't be read, this driver generates this ```BadData("Buffer length should be 8 not 0")```.

Here are some links to describe the issue:
https://github.com/gocql/gocql/pull/441
https://github.com/gocql/gocql/pull/441/files#diff-86b8b79c186e030c418e5b4843749915abbd25a8b34108a9982aa48811046cbfR842-R847

**Comment by psarna** [comment]
@cvybhu indeed, sounds like a specific case of our driver not being able to handle empty values. But I suppose that each type can have its own unique way of interpreting such an empty value, so it probably makes sense to solve the problem for each type (or class of similar types) independently
